### PR TITLE
nimble/audio: Add subgroups information for modify operation

### PR DESCRIPTION
**Pull request by szymon-czapracki**
Add subgroups info for modify source operation. Now metadata field can be handled in scan delegator during modify procedure.

### CODE REVIEW

**Comment by szymon-czapracki** [comment]
@MariuszSkamra @rymanluk FYI

File: nimble/host/audio/services/bass/include/services/bass/ble_audio_svc_bass.h [hunk 1]
```diff
@@ -324,6 +324,10 @@ struct ble_svc_audio_bass_operation {
             /** Number of subgroups */
             uint16_t num_subgroups;
 
+            /** Subgroup entries */
+            struct ble_svc_audio_bass_subgroup
+                subgroups[BLE_SVC_AUDIO_BASS_SUB_NUM_MAX];
+
             /** BIS Synchronisation of subgroups */
             uint32_t bis_sync[BLE_SVC_AUDIO_BASS_SUB_NUM_MAX];
```
Comment (MariuszSkamra) [inline nimble/host/audio/services/bass/include/services/bass/ble_audio_svc_bass.h:332]: this one should not be needed, as you can pass the bis sync in subgroups[i].bis_sync instead

Comment (szymon-czapracki) [reply nimble/host/audio/services/bass/include/services/bass/ble_audio_svc_bass.h]: LGTY now?

File: nimble/host/audio/services/bass/src/ble_audio_svc_bass.c [hunk 2]
```diff
@@ -424,20 +424,20 @@ ble_svc_audio_bass_add_source(uint8_t *data, uint16_t data_len, uint16_t conn_ha
 }
 
 static int
-check_bis_sync(uint16_t num_subgroups, const uint32_t *bis_sync_list)
+check_bis_sync(uint16_t num_subgroups, struct ble_svc_audio_bass_subgroup subgroups)
```
[the remark this answers is not in the document]
Comment (szymon-czapracki) [reply nimble/host/audio/services/bass/src/ble_audio_svc_bass.c:427]: Typo, sorry - fixed.

**Comment by MariuszSkamra** [comment]
LGTM :)

**Review by MariuszSkamra — APPROVED** [review]

**Review by sjanc — APPROVED** [review]

**Pull request merged**